### PR TITLE
Add service for getting expected output keys

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

File: Common/src/Storage/ITaskTable.cs [hunk 1]
```diff
@@ -300,8 +312,25 @@ public interface ITaskTable : IInitializable
...
+  {
+    try
+    {
+      return await GetTasksExpectedOutputKeys(new[]
+                                              {
+                                                taskId,
+                                              },
+                                              cancellationToken)
+                   .Select(tuple => tuple.expectedOutputKeys)
+                   .SingleAsync(cancellationToken)
```
Comment (lemaitre-aneo) [inline Common/src/Storage/ITaskTable.cs:326]: `SingleAsync` throws if sequence is empty, but Excpection would not give the right information (ie: missing task ID).

Comment (aneojgurhem) [reply Common/src/Storage/ITaskTable.cs:326]: There is a catch for this exception !

File: Common/src/gRPC/Services/GrpcTasksService.cs [hunk 2]
```diff
@@ -121,4 +120,44 @@ public class GrpcTasksService : Tasks.TasksBase
...
+               {
+                 await taskTable_.GetTasksExpectedOutputKeys(request.TaskId,
+                                                             context.CancellationToken)
+                                 .Select(r => new GetResultIdsResponse.Types.MapTaskResult
+                                              {
+                                                ResultIds =
+                                                {
+                                                  r.expectedOutputKeys,
+                                                },
+                                                TaskId = r.taskId,
```
Comment (wkirschenmann) [inline Common/src/gRPC/Services/GrpcTasksService.cs:141]: Just for coherence with the object name, I'd put the TaskId before the ResultIds

File: Adaptors/MongoDB/src/TaskTable.cs [hunk 3]
```diff
@@ -626,6 +624,24 @@ await foreach (var taskId in taskCollection.AsQueryable(sessionHandle)
...
 
+  public IAsyncEnumerable<(string taskId, IEnumerable<string> expectedOutputKeys)> GetTasksExpectedOutputKeys(IEnumerable<string> taskIds,
+                                                                                                              CancellationToken   cancellationToken = default)
+  {
+    using var activity       = activitySource_.StartActivity($"{nameof(GetTasksExpectedOutputKeys)}");
+    var       sessionHandle  = sessionProvider_.Get();
+    var       taskCollection = taskCollectionProvider_.Get();
+
+    return taskCollection.AsQueryable(sessionHandle)
+                         .Where(tdm => taskIds.Contains(tdm.TaskId))
```
[the remark this answers is not in the document]
Comment (lemaitre-aneo) [reply Adaptors/MongoDB/src/TaskTable.cs:635]: The question was: if we have a `IMongoQueryable`, are we sure that the function used is the one from mongo, and not the one from IAsyncEnumerable. I'm fine to defer this discussion.

**Review by lemaitre-aneo — DISMISSED** [review]

**Review by lemaitre-aneo — APPROVED** [review]

**Review by wkirschenmann — APPROVED** [review]

**Pull request merged**